action/envelope.go

```diff
@@ -18,7 +18,7 @@ import (
 type (
 	// Envelope defines an envelope wrapped on action with some envelope metadata.
 	Envelope interface {
-		Version() uint32
+		CommonAction() AbstractAction
```

Comment:
so far, keep this as a struct
in the next PR, will convert to an interface (and have the 4 types of transactions implement it)

Reply:
can we just inherit the `AbstractAction`? I didn't see the necessity of making it as a method

Reply:
yes, removed in latest code